process/smartContract/process.go

```diff
@@ -480,6 +488,7 @@ func (sc *scProcessor) processVMOutput(
 
 	totalGasRefund := big.NewInt(0)
 	totalGasRefund = totalGasRefund.Add(vmOutput.GasRefund, vmOutput.GasRemaining)
+	sc.gasHandler.SetGasRefunded(totalGasRefund.Uint64())
```

Comment:
use AddGasRefunded

Reply:
done

Reply:
refactored GasConsumption to hold for both of these values, gasConsumed and gasRefunded, a map of hash and value